gslib/utils/copy_helper.py

```diff
@@ -3205,29 +3216,34 @@ def _ValidateAndCompleteDownload(logger,
       if gzip_fp:
         gzip_fp.close()
 
-    os.unlink(file_name)
-    file_name = final_file_name
+    os.unlink(temporary_file_name)
 
   if not digest_verified:
     try:
       # Recalculate hashes on the unzipped local file.
       local_hashes = _CreateDigestsFromLocalFile(gsutil_api.status_queue,
-                                                 hash_algs, file_name, src_url,
-                                                 src_obj_metadata)
+                                                 hash_algs,
+                                                 unzipped_temporary_file_name,
+                                                 src_url, src_obj_metadata)
       _CheckHashes(logger, src_url, src_obj_metadata, final_file_name,
                    local_hashes)
       DeleteDownloadTrackerFiles(dst_url, api_selector)
     except HashMismatchException:
       DeleteDownloadTrackerFiles(dst_url, api_selector)
       if _RENAME_ON_HASH_MISMATCH:
-        os.rename(file_name, file_name + _RENAME_ON_HASH_MISMATCH_SUFFIX)
+        os.rename(
+            unzipped_temporary_file_name,
+            unzipped_temporary_file_name + _RENAME_ON_HASH_MISMATCH_SUFFIX)
       else:
-        os.unlink(file_name)
+        os.unlink(unzipped_temporary_file_name)
       raise
 
-  if file_name != final_file_name:
+  if unzipped_temporary_file_name != final_file_name:
     # Data is still in a temporary file, so move it to a permanent location.
-    os.rename(file_name, final_file_name)
+    os.rename(unzipped_temporary_file_name, final_file_name)
```

Comment:
Shouldn't the renaming happen after the stet transform?

Reply:
Oh, yikes, I did all the setup, but forgot to make the actual change.